python/lsst/pipe/base/connections.py

```diff
@@ -493,15 +493,8 @@ class DeferredDatasetRef:
 
     datasetRef: DatasetRef
 
-    @property
-    def datasetType(self) -> DatasetType:
-        """The dataset type for this dataset."""
-        return self.datasetRef.datasetType
-
-    @property
-    def dataId(self) -> DataCoordinate:
-        """The data ID for this dataset."""
-        return self.datasetRef.dataId
+    def __getattr__(self, name: str) -> Any:
+        return getattr(self.datasetRef, name)
```

Comment:
This change is because you needed some other method on datasetRef? Or because you wanted to simplify the code a bit?

Reply:
Yes, there was some previously-unused code in the mocking stuff that needed `id` that was uncovered by the other change.